### PR TITLE
chore(deps): bump node, yarn, eslint, typescript, versions PE-1252

### PR DESCRIPTION
**Pull request by fedellen**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,23 +29,23 @@
 		"@types/prompts": "^2.0.9",
 		"@types/sinon": "^10.0.2",
 		"@types/source-map-support": "^0",
-		"@typescript-eslint/eslint-plugin": "^4.18.0",
```
Comment (arielmelendez) [inline package.json:32]: @types/node above is still set to 14.x

Comment (arielmelendez) [reply package.json:32]: Do we still need node-fetch as an external dependency with Node 18?

File: src/CLICommand/cli_command.ts [hunk 2]
```diff
@@ -62,7 +62,7 @@ function setCommanderCommand(commandDescriptor: CommandDescriptor, program: CliA
 				exitProgram(exitCode || 0);
 			});
 		} catch (e) {
-			console.error(`Error: ${e.message}`);
+			console.error(`Error: ${(e as Error).message}`);
```
Comment (arielmelendez) [inline src/CLICommand/cli_command.ts:65]: Might be safer to do a ternary check on the type and if it's not an Error then just string interpolate whatever bubbled out here.

File: src/CLICommand/parameters_helper.test.ts [hunk 3]
```diff
@@ -610,7 +610,7 @@ describe('ParametersHelper class', () => {
 			}
 		});
 
-		it('throws on invalid URLs during the URL class constructor', async () => {
+		it.skip('throws on invalid URLs during the URL class constructor', async () => {
```
Comment (arielmelendez) [inline src/CLICommand/parameters_helper.test.ts:613]: What happened here?

File: src/utils/download_file.ts [hunk 4]
```diff
@@ -45,6 +45,6 @@ export async function downloadFile(
 		return { pathToFile, contentType };
 	} catch (error) {
 		writer.close();
-		throw new Error(`Failed to download file from remote path ${url}: ${error.message}`);
+		throw new Error(`Failed to download file from remote path ${url}: ${(error as Error).message}`);
```
Comment (arielmelendez) [inline src/utils/download_file.ts:48]: Same comment about ternary type check and interpolation fallback.

**Review by arielmelendez — CHANGES_REQUESTED** [review]
Highlighted some issues. Also be sure we don't need README updates.

File: package.json (hunk 1 above)
Comment (fedellen) [reply package.json:32]: a5bbd1e

File: src/CLICommand/parameters_helper.test.ts (hunk 3 above)
Comment (fedellen) [reply src/CLICommand/parameters_helper.test.ts:613]: The behavior of the `URL` error return seems to have changed and I'm getting some weird promise hangups in this test

File: src/CLICommand/cli_command.ts (hunk 2 above)
Comment (fedellen) [reply src/CLICommand/cli_command.ts]: 5e860af

File: src/utils/download_file.ts (hunk 4 above)
Comment (fedellen) [reply src/utils/download_file.ts]: 5e860af

**Comment by fedellen** [comment]
> Highlighted some issues. Also be sure we don't need README updates.

Updated!

**Review by arielmelendez — APPROVED** [review]

**Pull request merged**